adam/experiment/observer.py

```diff
@@ -874,7 +874,7 @@ def observe(  # pylint: disable=unused-argument
             semantic_node, linguistic_description = semantic_to_description
             output_dict[OUTPUT_LANGUAGE].append(
                 self._convert_to_output_format(
-                    idx,
+                    semantic_node.original_node_id if semantic_node.original_node_id is not None else idx + len(predicted_scene_description.semantics_to_descriptions),  # type: ignore
```

Comment:
This is a complicated enough expression that it probably deserves either (1) a comment or (2) a separate function. Using the original node ID makes sense, but I'm not sure I understand the fallback (at least not from the immediate diff context).

Reply:
The new comment looks good to me. Thanks!